app/src/main/java/com/hedvig/app/feature/embark/passages/previousinsurer/retrieveprice/RetrievePriceViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.hedvig.app.feature.embark.passages.previousinsurer.retrieveprice
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.hedvig.app.feature.settings.Market
+import com.hedvig.app.feature.settings.MarketManager
+import com.hedvig.app.util.validateNationalIdentityNumber
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class RetrievePriceViewModel(
+    marketManager: MarketManager,
+    private val startDataCollectionUseCase: StartDataCollectionUseCase
+) : ViewModel() {
+
+    private val _viewState = MutableStateFlow(ViewState(market = marketManager.market))
+    val viewState: StateFlow<ViewState> = _viewState
+
+    fun onRetrievePriceInfo() {
+        if (viewState.value.inputError != null) {
+            return
+        }
+
+        viewModelScope.launch {
+            _viewState.update { it.copy(isLoading = true) }
+            val result = startDataCollectionUseCase.startDataCollection(
+                personalNumber = viewState.value.input,
+                insuranceProvider = "ICA"
```

Comment:
This will be part of [APP-1086](https://hedvig.atlassian.net/jira/software/c/projects/APP/boards/6?modal=detail&selectedIssue=APP-1086&quickFilter=51)